### PR TITLE
[clang][Sema] Subclass `-Wshorten-64-to-32` under `-Wimplicit-int-conversion`

### LINKED ISSUE #69444: [clang][Sema] Question: Why is `-Wshorten-64-to-32` disjoint from `-Wimplicit-int-conversion`?
An interesting case was encountered where conversions from larger bit-width integers to smaller bit-width ones is *generally* reported by `-Wimplicit-int-conversion` (32 to 16, 16 to 8, 128 to 64, etc.) **except** for 64 to 32, which was given its own warning: `-Wshorten-64-to-32`.

See http://godbolt.org/z/6q4GMPsxs. Lines 19 (`uint64_t` -> `uint32_t`) and 35 (`int64_t` -> `int32_t`) are not warned about, unless _"Compiler 2"_'s warning is enabled.

Although `-Weverything` enables both of them at the same time (and neither is enabled by default, AFAIK), it is strange that 64->32 is such a specific dedicated case.

Unfortunately, all I could track down was commit 263a48b781f360fc75f4b95fb9abbb866209cb21 where this warning was introduced, and that is **ancient**, but history runs dry at that point.

> CC @dwblaikie, @rjmccall. Your names appeared in the relevant code path.

----

**What would be the cons against enabling `-Wshorten-64-to-32` under `-Wimplicit-int-conversion` implicitly?**

### PR DESCRIPTION
**Pull request by whisperity**
Although "implicit int conversions" is supposed to be a superset containing the more specific "64-to-32" case, previously they were a disjoint set, only enabled in common in the much larger `-Wconversion`.

Closes #69444.

`diagtool tree` diff:

```diff
   -Wconversion
     # ...
     -Wfloat-conversion
       -Wfloat-overflow-conversion
       -Wfloat-zero-conversion
-    -Wshorten-64-to-32
     -Wint-conversion
     -Wimplicit-int-conversion
+      -Wshorten-64-to-32
       -Wobjc-signed-char-bool-implicit-int-conversion
     -Wimplicit-float-conversion
       -Wimplicit-int-float-conversion
         -Wimplicit-const-int-float-conversion
       -Wobjc-signed-char-bool-implicit-float-conversion
     # ...
```

### CODE REVIEW

**Comment by dwblaikie** [comment]
Can't seem to load the image - and generally a copy/paste of the text is more usable for everyone than a screenshot. If you could include the copy/pasted text, that'd be handy, thanks!

File: clang/docs/ReleaseNotes.rst [hunk 1]
```diff
@@ -149,7 +149,12 @@ Improvements to Clang's diagnostics
   prints.
 
 - Clang now diagnoses member template declarations with multiple declarators.
-- Clang now diagnoses use of the ``template`` keyword after declarative nested name specifiers.
+
+- Clang now diagnoses use of the ``template`` keyword after declarative nested
+  name specifiers.
+
+- Added `-Wshorten-64-to-32` diagnostics to enable when
+  `-Wimplicit-int-conversion` is enabled.
```
Comment (AaronBallman) [inline clang/docs/ReleaseNotes.rst:157]:
```suggestion
- The ``-Wshorten-64-to-32`` diagnostic is now grouped under ``-Wimplicit-int-conversion`` instead
   of ``-Wconversion``. Fixes `#69444 <https://github.com/llvm/llvm-project/issues/69444>`_.
```

**Review by AaronBallman — COMMENTED** [review]
LGTM aside from a minor change to the release notes, thank you for this improvement!

**Comment by rjmccall** [comment]
This looks right to me, thanks.

**Pull request merged**